### PR TITLE
Increase TextField's minimum height from 40 to 48

### LINKED ISSUE #41200: TextFields don't meet minimum accessibility height (48 px) with default style (just 43)
Internal: b/141526836

Minimum a11y height requirements as per Material is 48dp whereas default style gives a height of 43.

### PR DESCRIPTION
**Pull request by justinmc**
## Description

Here are some screenshots for comparison.  The red square is just a Container of size 48x48 as a reference.  The green is the background of the TextField.

| Before | After |
| -------------- | ------------ |
| flutter_18 | flutter_15 |
| flutter_19 | flutter_16 |
| flutter_20 | flutter_21 |

## Related Issues

Closes https://github.com/flutter/flutter/issues/41200

## Tests

I added two tests to verify the minimum height is preserved for the default case and the case where text is very small.  I also updated the numbers in tons of tests to reflect the new dimensions.

## Breaking Change

This will break many visual diff tests due to the change in size of default TextFields and the change in the location of the baseline.  I will follow the [breaking changes guildelines](https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes) after getting code review.

### CODE REVIEW

**Comment by justinmc** [comment]
Some things I need to check on:
  * Are outlined text fields affected by this?
  * What happens if the textfield is forced to be very tall? Baseline alignment should work the same before and after this PR.

File: packages/flutter/test/material/text_field_test.dart
Comment (justinmc) [inline packages/flutter/test/material/text_field_test.dart]: This seems weird... I don't know why it isn't receiving a tap with the previous line now.  If anyone has a hunch please let me know.

**Comment by justinmc** [comment]
Goldens are being updated here: https://github.com/flutter/goldens/pull/58

**Review by HansMuller — COMMENTED** [review]
This looks good and its nice to see that such a big change can be accomplished with such a small change.

I think we might want to make isDense:true an exception to the min height = 48; just restore the original behavior in that case

**Comment by jonahwilliams** [comment]
This should probably utilize https://api.flutter.dev/flutter/material/MaterialTapTargetSize-class.html in some way to avoid extra padding where it doesn't matter, like desktop

**Comment by justinmc** [comment]
I discussed the MaterialTapTargetSize offline and we decided to leave it out of this PR in anticipation of some upcoming work for Material.

**Comment by justinmc** [comment]
Draft breaking change email:

> I'm proposing to increase the default height of TextField by eight pixels in order to comply with the Material spec's minimum interactive size guideline.
> 
> Issue: https://github.com/flutter/flutter/issues/41200
> PR: https://github.com/flutter/flutter/pull/42449
> 
> Currently, TextFields are 40px high with default settings.  This change would set a minimum height of 48px for all TextFields where the isDense flag on InputDecorator is not true.  Any field whose height is calculated to be less than 48px would be bumped up to 48 and its content vertically centered.  Smaller TextFields could still be created with the isDense flag set to true.
> 
> Anyone that currently has a TextField that is below 48px high in their app, including those just using the default TextField settings, will see the size of these fields increase.  These affected apps will need to update any visual diff tests covering these fields, and may also need to adjust their layouts to accommodate the larger fields.
> 
> If someone wishes to keep their TextFields at a height of less than 48px, they may do so by using the isDense flag.
> 
> Please let me know if you have any trouble migrating and need some help!  I'm reachable at this email address or on Github in the given PR (@justinmc).
> 
> If anyone objects to this change or would like to discuss further, please comment in the linked PR.
> 
> 
> Thanks,
> 
> Justin

**Comment by justinmc** [comment]
@Piinks If I still see goldens failures after merging my goldens PR, does that mean it's my fault and the goldens I added don't match?

Sorry, I'm forgetting if there's anything else I need to do to update the goldens here.

**Comment by Piinks** [comment]
Have the version numbers been incremented to match the golden file changes?

**Comment by justinmc** [comment]
That's it, I knew I forgot something.  Thank you!

**Comment by Piinks** [comment]
https://github.com/flutter/flutter/pull/43371 has changed golden file testing, which will need a few minor adjustments here. When you resolve the merge conflicts, the `goldens.version` file will no longer exist.
It looks like this change affects already existing tests, so in order for them to pass pre-submit here, an ignore will have to be put in place for the affected tests in [Flutter Gold](https://flutter-gold.skia.org/?query=source_type%3Dflutter). 
As far as I can tell it looks like the affected tests are

 - text_field_cursor_test.material.0
- text_field_cursor_test.material.1
- text_field_opacity_test.0

Take a look at the updated [Writing a golden file test for package:flutter](https://github.com/flutter/flutter/wiki/Writing-a-golden-file-test-for-package:flutter), and let me know if I can help migrate this change to the new system. :)

**Review by HansMuller — APPROVED** [review]
LGTM

**Comment by Piinks** [comment]
Can we wait to land this until tomorrow? 

- The duration of the ignore should be set to cover the amount of time it will take to be triaged. Feel free to edit it to something conservative (a day or two is totally fine).

Also, there are performance improvements that should land by tomorrow AM that will improve the change going through: 
- https://skia-review.googlesource.com/c/buildbot/+/251459
- https://github.com/flutter/flutter/pull/43748

**Comment by justinmc** [comment]
Yeah that's fine with me!  This PR has a breaking change so I don't want to rush it anyway.  I will hold off until tomorrow.  I've updated the expiration to 2 days.

**Comment by Piinks** [comment]
Gold updates have landed, and I renewed the ignore on the dashboard. :)

**Pull request merged**
